### PR TITLE
Implement SupportSQLite* interfaces to be compatible with third party libraries

### PR DESCRIPTION
**Pull request by gabrielittner**
Notes:
- bumps min sdk 14
- bumps supportlib to 26.1.0

Questions:
- [x] Naming of `RequerySQLiteOpenHelperFactory`. Requery is not used in any other name but it I think it makes sense to have something unique here. The one that returns a wrapped framework `SQLiteOpenHelper` is called `FrameworkSQLiteOpenHelperFactory`.
- [x] `SupportSQLiteDatabase` brings some methods that are slightly different than the `SQLiteDatabase` ones. Should the old ones be deprecated or removed? (not sure what you're policy here is)
  - [x] `rawQuery(String, Object[])` identical to `query(String, Object[])`
  - [x] Depending on how the last one is handled: `SupportSQLiteDatabase` only uses query as name not rawQuery. Should the other rawQuery methods be renamed?
  - [x] `delete(String, String, String[])` vs `delete(String, String, Object[])`
  - [x] `update(String, ContentValues, String, String[], int)` vs `update(String, int, ContentValues, String, Object[])`

Todos (want to get the question part out of the way first):
- [x] add javadoc to new SQLiteDatabase methods
- [x] update README
- [x] update tests

### CODE REVIEW

**Comment by npurushe** [comment]
Hi thanks for the PR! for your questions/comments:

1) minSdkVersion increase to 14, I believe there were some users who still have to support < 14, need to verify that but otherwise should be ok
2) `RequerySQLiteOpenHelperFactory` this class name is ok
3) `SupportSQLiteDatabase` is part of new library components which have a different goal from the normal support libraries. The main goal of the library is to support an api compatible with `android.database.sqlite.SQLiteDatabase`. Renaming or removing methods would break existing users usage of the library so that has to be avoided. Similarly only methods deprecated in `android.database.sqlite.SQLiteDatabase` should be marked deprecated in this library. Adding new methods is ok though, looking through your PR the additions shouldn't cause a problem.

File: sqlite-android/build.gradle [hunk 1]
```diff
@@ -18,14 +18,11 @@ gradle.projectsEvaluated {
...
-    compileSdkVersion 25
-    buildToolsVersion "25.0.2"
+    compileSdkVersion 26
+    buildToolsVersion "26.0.1"
 
     defaultConfig {
-        minSdkVersion 9
-        targetSdkVersion 25
-        versionCode 1
-        versionName project.version
```
Comment (npurushe) [inline sqlite-android/build.gradle]: is there a reason this is removed? the version name should be available as part of the library build info I believe

Comment (gabrielittner) [reply sqlite-android/build.gradle]: The versionName removal was accidental, added it back.

**Comment by gabrielittner** [comment]
From what I can see the SQLiteDatabase methods are not tested themselves and only used to test underlying database and Cursor functionality. Do any of the newly added methods need a test?

**Comment by npurushe** [comment]
SQLiteDatabase  is tested mostly by `DatabaseGeneralTest` in the test code, Adding tests for the added update/delete methods would be helpful. Everything else looks straight forward thanks!

**Comment by gabrielittner** [comment]
Added tests

**Comment by npurushe** [comment]
Great thanks!

**Pull request merged**